WordPress/gutenberg
# Query Pagination: do not render pagination container if there's no pagination

## Description
This PR adds a render callback to the query pagination block. This way, the pagination container element is only rendered if pagination exists.

I think rendering an empty container in most cases is harmless, but is it better to actually not add anything to the DOM if there isn't content present? There are probably other cases where this happens in FSE blocks..

This is an attempt to solve https://github.com/Automattic/themes/issues/4268 where we style the wrapper only when pagination is present.

## How has this been tested?
1. Create a post or index template with the following query block markup on a site with fewer than five posts:

```

<main class="wp-block-query">
	
		
		
		
		
	
	
	
	
	
	
</main>

``` 

2. Verify that no pagination container is rendered 
3. Verify relevant block attributes ("wide" in the above example) are present on the wrapper

## Screenshots 

## Types of changes
Bug fix? 

## Checklist:
